### DISCUSSION TITLE
How to call to an existing bucket ? ( JDBC integrations psql)

### DISCUSSION THREAD

**Discussion by talo-sgh** [vladimir-bukhtoyarov/bucket4j#352] [Q&A] [answered]
Need help understanding how I get an existing bucket. 
Do I need to build BucketProxy each time?

Thanks

**Comment by vladimir-bukhtoyarov** [comment, marked as answer]
In most cases Bucket Proxy is cheap wrapper arround primary key, so there are no differences between creation proxy each time and storing once created proxy somewhere, caching the proxy will not allow to save somesing significant. So you should prefer the way that makes your code more clean. If build use and forget makes your code simpler then use this way. But if you have deal with a few buckets that are well known at development time then it is obviosly that all thes buckets can be created at application start time and just reused during application life time.

Ptoxy manager in opposite to BucketProxy should be created only once.
